skqulacs/qnn/regressor.py

```diff
@@ -49,10 +49,11 @@ class QNNRegressor:
     do_y_scale: bool = field(default=True)
     x_norm_range: float = field(default=1.0)
     y_norm_range: float = field(default=0.7)
+    observables_str: List[str] = field(init=True, default_factory=list)
```

Comment:
Could you explain an advantage of `observables_str`? I was not able to figure out.

Reply:
I think that before you could not specify the observable when creating the Regressor, it was set as Z_i based in the i dimensions of the output. In this way you can declare an arbitrary observable (or set of observables)

Reply:
I got it! But this is still complex...
How about passing `init=True` to `field` for `observables` so users can pass observables?
And move the initialization of default observables and initialize them only if `self.observable` is `[]`(when observables are not passed in constructor)?
```python
@dataclass
class QNNRegressor:
    # cut
    observables = field(init=True, defalut_factory=list)

    def __post_init__(self) -> None:
        # cut
        if len(self.observables) == 0:
            for i in range(n_outputs):
                observable = Observable(n_qubit)
                observable.add_operator(1.0, f"Z {i}")
                self.observables.append(observable)
```

Reply:
My philosophy was more that a user would prefer writing a string instead of looking at the documentation for how to define the operators.
But I think your solution is better, my problems are that I don't know then how to treat:
`        for h in range(len(x_scaled)):
            backobs = Observable(self.n_qubit)
            if self.n_outputs >= 2:
                for i in range(len(self.observables_str)):
                    backobs.add_operator(
                        2 * (-y_scaled[h][i] + mto[h][i]) / self.n_outputs,
                        self.observables_str[
                            i
                        ],`
and similar (in Regressor.py) because now I already have some files with Observables but I don't know which are they. Probably is just my lack of knowledge of qulacs operators 😂

Reply:
the current released version does `(-y_scaled[h][i] + mto[h][i]) / self.n_outputs, f"Z {i}"` that you see has very little freedom

Reply:
Ok, I agree with that passing string representations of an observable is easier.
The code you mention is the one in line 197-206, right? Currently we do not have any concise API in qulacs to construct an observable from existing one with a different coefficient. So we accept your code as a work around, and we are going to discuss how to deal with it in another issue in qulacs.

Reply:
I understand and I agree that it would be a good issue in qulacs. 
Yes, the code is from the position you mention :)